src/grpc_server.cc

```diff
@@ -1340,6 +1340,224 @@ CommonHandler::SetUpAllRequests()
       inference::TraceSettingRequest, inference::TraceSettingResponse>(
       "Trace", 0, OnRegisterTrace, OnExecuteTrace, false /* async */, cq_);
 
+  //
+  //  Log Settings
+  //
+
+  auto OnRegisterLogging =
+      [this](
+          grpc::ServerContext* ctx, inference::LogSettingsRequest* request,
+          grpc::ServerAsyncResponseWriter<inference::LogSettingsResponse>*
+              responder,
+          void* tag) {
+        this->service_->RequestLogSettings(
+            ctx, request, responder, this->cq_, this->cq_, tag);
+      };
+
+  auto OnExecuteLogging = [this](
+                              inference::LogSettingsRequest& request,
+                              inference::LogSettingsResponse* response,
+                              grpc::Status* status) {
+
+#ifdef TRITON_ENABLE_LOGGING
+    TRITONSERVER_Error* err = nullptr;
+    std::string log_file_path = LOG_OUT_FILE;
+    bool log_info_status = LOG_INFO_IS_ON;
+    bool log_warn_status = LOG_WARNING_IS_ON;
+    bool log_error_status = LOG_ERROR_IS_ON;
+    uint32_t verbose_level = LOG_VERBOSE_LEVEL;
+    triton::common::Logger::Format log_format_final = LOG_FORMAT;
+    std::string log_format_parse = LOG_FORMAT_STRING;
```

Comment:
I don't think you need to declare these variables early, when you are preparing the response, you can just invoke the macro, i.e.: 
```
(*response->mutable_settings())["log_file"].set_string_param(LOG_OUT_FILE);
```